ema_workbench/em_framework/optimization.py

```diff
@@ -512,28 +667,72 @@ class ArchiveLogger(AbstractConvergenceMetric):
     base_filename : str, optional
 
 
-    TODO:: put it in a tarbal instead of dedicated directory
-
     """
 
-    def __init__(self, directory, decision_varnames, outcome_varnames, base_filename="archive"):
+    def __init__(
+        self, directory, decision_varnames, outcome_varnames, base_filename="archives.tar.gz"
+    ):
         super().__init__("archive_logger")
+
+        # FIXME how to handle case where directory already exists
         self.directory = os.path.abspath(directory)
+        self.temp = os.path.join(self.directory, "tmp")
+        os.mkdir(self.temp)
+
         self.base = base_filename
         self.decision_varnames = decision_varnames
         self.outcome_varnames = outcome_varnames
-        self.index = 0
+        self.tarfilename = os.path.join(self.directory, base_filename)
+
+        # self.index = 0
 
     def __call__(self, optimizer):
-        self.index += 1
+        archive = to_dataframe(optimizer.result, self.decision_varnames, self.outcome_varnames)
+        archive.to_csv(os.path.join(self.temp, f"{optimizer.nfe}.csv"))
 
-        fn = os.path.join(self.directory, f"{self.base}_{self.index}.csv")
+    def reset(self):
+        # FIXME what needs to go here?
+        pass
 
-        archive = to_dataframe(optimizer, self.decision_varnames, self.outcome_varnames)
-        archive.to_csv(fn)
+    def get_results(self):
+        with tarfile.open(self.tarfilename, "w:gz") as z:
+            z.add(self.temp, arcname=os.path.basename(self.temp))
+
+        shutil.rmtree(self.temp)
+        return None
+
+    @classmethod
+    def load_archives(cls, filename):
+        """load the archives stored with the ArchiveLogger
+        Parameters
+        ----------
+        filename : str
+                   relative path to file
+        Returns
+        -------
+        dict with nfe as key and dataframe as vlaue
+        """
+
+        archives = {}
+        with tarfile.open(os.path.abspath(filename)) as fh:
+            for entry in fh.getmembers():
+                if entry.name.endswith("csv"):
+                    key = entry.name.split("/")[1][:-4]
+                    archives[int(key)] = pd.read_csv(fh.extractfile(entry))
+        return archives
 
 
 class OperatorProbabilities(AbstractConvergenceMetric):
+    """Operator Probabiliy convergence tracker for use with
+    auto adaptive operator selection
```

Comment:
Could you extend this docstring a tiny bit? Maybe what is does in two sentences.

Reply:
What exactly do you feel is missing?

Reply:
When/why you use this maybe? Or what you can practically do with this?

I also just tumble over a very high ratio of complex words in one sentence

Reply:
I added some more text to the documentation